stdlib/ballerina-builtin/src/main/ballerina/runtime/Package.md

```diff
@@ -0,0 +1,100 @@
+## Package overview
+
+This package includes functions to interact with the runtime, the invocation context and to manage errors.
+
+### Invocation Context
+
+The Invocation Context is a data holder which is created per request, and preserved for a single request-response flow. 
+The Invocation Context comprises of a unique id, UserPrincipal instance which includes user details and an AuthContext 
+instance which has the authentication related details if available.
+
+### Errors
+
+Error types NullReferenceException and IllegalStateException, which wraps the error type 
+defined in ballerina/builtin package, are included in the runtime package. Furthermore, there are utility methods to 
+retrieve the current call stack and the particular call stack frame for an error. 
+
+Additionally, the runtime package also contains util methods to halt a thread (sleep) for a given period of time and 
+to lookup properties from the runtime context.
+
+### Samples
+
+The following sample shows how to access the Invocation Context, set data to it and access the same.
+```ballerina
+import ballerina/runtime;
+import ballerina/io;
+
+// set data to the Invocation Context
+// set the username ‘tom’ as the user name
+runtime:getInvocationContext().userPrincipal.username = "tom";
+// set claims
+map claims = { email: "tom@ballerina.com", org: "wso2" };
+runtime:getInvocationContext().userPrincipal.claims = claims;
+// set scopes
+string[] scopes = ["email", "profile"];
+runtime:getInvocationContext().userPrincipal.scopes = scopes;
+// set auth scheme
+runtime:getInvocationContext().authContext.scheme = "jwt";
+// set auth token
+runtime:getInvocationContext().authContext.authToken = "abc.pqr.xyz";
+
+// retrieve data from the invocation context
+// retrieve user name
+string userName = runtime:getInvocationContext().userPrincipal.username;
+io:println(userName);
+// retrieve claims
+map retrievedClaims = <map>runtime:getInvocationContext().userPrincipal.claims;
+io:println(retrievedClaims);
+// retrieve scopes
+string[] retrievedScopes = runtime:getInvocationContext().userPrincipal.scopes;
+io:println(retrievedScopes);
+// retrieve auth scheme
+string authScheme = runtime:getInvocationContext().authContext.scheme;
+io:println(authScheme);
+// retrieve auth token
+string token = runtime:getInvocationContext().authContext.authToken;
+io:println(token);
+```
+
+The following sample shows how to sleep a worker thread for a given time period.
```

Comment:
**Suggestion:**
"The following sample shows how to set the mode of a worker thread to 'sleep' for a given time period."

Reply:
I don't think mode is the correct term here, how about: 
"The following sample shows how to halt a `worker` thread for a given time period."

Reply:
Ok